### PR TITLE
#91 Add consumer endpoint

### PR DESCRIPTION
**Pull request by nikkite99**

### CODE REVIEW

File: backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@Service
+public class ConsumerService {
+
+    public ConsumerStats getConsumerStatsByTopic(TopicStats topicStats, String consumer) {
+        return topicStats.getSubscriptions()
+                .values()
+                .stream().flatMap(
+                        subscriptionStats -> subscriptionStats.getConsumers().stream()
+                )
```
Comment (jnsrnhld) [inline backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java:20]:
```suggestion
        return topicStats.getSubscriptions()
                .values().stream()
                .flatMap(subscriptionStats -> subscriptionStats.getConsumers().stream())
```
Just my personal flavor.

**Review by jnsrnhld — APPROVED** [review]
Just some formatting stuff.

File: backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package de.amos.apachepulsarui.service;
+
+import org.apache.pulsar.common.policies.data.ConsumerStats;
+import org.apache.pulsar.common.policies.data.TopicStats;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConsumerService {
```
Comment (ahaverkamp) [inline backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java:13]: I like that you created a new service class, makes everything clearer. I was wondering if we should do the same for the producer stuff?

**Review by ahaverkamp — APPROVED** [review]

File: backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java (hunk 2 above)
Comment (nikkite99) [reply backend/src/main/java/de/amos/apachepulsarui/service/ConsumerService.java:13]: I should be you are right. I was a bit busy so ill do it in the next sprint :)

**Pull request merged**